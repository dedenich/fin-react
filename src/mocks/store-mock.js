import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const store = mockStore({
  admissions: [],
  remains: 0,
  collectedShare: 0,
  collectedAmount: 0,
  aim: {
    description: ``,
    amount: `0`,
  },
  isAimAdded: false,
});

export {store};
