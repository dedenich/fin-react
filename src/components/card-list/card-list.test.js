import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import {store as mockStore} from '../../mocks/store-mock.js';

import CardList from './card-list.jsx';

it(`CardList renders correctly`, () => {
  const tree = renderer
  .create(
      <Provider store={mockStore}>
        <CardList
          isAdmissionsCard={true}
          isContentShown={true}
        />
      </Provider>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
