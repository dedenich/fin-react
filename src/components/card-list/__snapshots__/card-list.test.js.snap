// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders correctly 1`] = `
Array [
  <button
    className="close_button "
  />,
  <ul
    className="card-list"
  />,
]
`;
